### PR TITLE
Add Modbus communication

### PR DESCRIPTION
**Pull request by AndreaRicchi**
This pr will add Modbus communication, linking the GUI interaction with the Modbus server.

Since the pr is large, I suggest the reviewers go commit by commit to better understand the development process.

### CODE REVIEW

File: src/gui/MainWindow.hpp [hunk 1]
```diff
@@ -41,7 +41,7 @@ class MainWindow : public QMainWindow
     QString m_importFilePath;
 
     std::unique_ptr<ModbusCom> m_modbus;
-    std::vector<std::shared_ptr<MdiChild>> m_mdiChilds;
+    std::vector<MdiChild *> m_mdiChilds;
```
Comment (marsiliano) [inline src/gui/MainWindow.hpp:44]: Raw Pointer?

**Review by marsiliano — CHANGES_REQUESTED** [review]

File: src/gui/MainWindow.hpp (hunk 1 above)
Comment (AndreaRicchi) [reply src/gui/MainWindow.hpp:44]: Unfortunately, there is no other way. Using smart pointers will cause multiple deallocations, from the smart pointer itself and from the Qt memory management, ending in segmentation faults.
In other words, this vector of pointers will be deallocated upon the MainWindow destruction.

Comment (marsiliano) [reply src/gui/MainWindow.hpp:44]: Open a bug to Qt

**Review by marsiliano — APPROVED** [review]

File: src/gui/MainWindow.cpp [hunk 2]
```diff
@@ -31,6 +31,16 @@ MainWindow::MainWindow(QWidget *parent) :
 
     connect(this, &MainWindow::importFinished, this,
             &MainWindow::createConfigView);
+
+    m_modbus.reset(new ModbusCom(this));
```
Comment (guerinoni) [inline src/gui/MainWindow.cpp]: Why not just `m_modbus = std::make_unique<ModbusCom>();` using `this` as parameter Mainwindow try to delete this class as child. I prefer using Qt memory management only with GUI element

File: src/serialcom/ModbusCom.cpp [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+{
+    if (error == QModbusDevice::NoError || !m_modbusDevice) {
+        return;
+    }
+
+    qWarning() << QStringLiteral("ModbusCom: Error occurred: %1 (code 0x%2)")
+                      .arg(errorString)
+                      .arg(error);
+
+    switch (error) {
```
Comment (guerinoni) [inline src/serialcom/ModbusCom.cpp:105]: this is apparently useless

File: src/src.pro [hunk 4]
```diff
@@ -4,5 +4,6 @@ CONFIG   += ordered
 SUBDIRS +=  \
```
Comment (guerinoni) [inline src/src.pro]: please, keep vertical alignment of backslash

**Review by guerinoni — CHANGES_REQUESTED** [review]

File: src/serialcom/ModbusCom.cpp (hunk 3 above)
Comment (AndreaRicchi) [reply src/serialcom/ModbusCom.cpp:105]: More needs to be done, attaching the error event to the GUI.

**Pull request merged**